prime-router/src/main/kotlin/fhirengine/utils/CompareFhirData.kt

```diff
@@ -217,7 +214,16 @@ class CompareFhirData(
         }
 
         if (!primitiveResult.passed) {
-            val msg = "FAILED: Property $primitiveIdPath $primitiveTypePath did not match"
+            val msg = if (expectedPrimitive is PrimitiveType<*>) {
+                val actualPrimitivesValues = actualPrimitive
+                    .filterIsInstance<PrimitiveType<*>>()
+                    .joinToString { it.asStringValue() }
+                "FAILED: Property $primitiveIdPath $primitiveTypePath did not match. " +
+                    "expected=${expectedPrimitive.asStringValue()}, actuals=$actualPrimitivesValues"
```

Comment:
Really like this update here!

```suggestion
                    "expected=${expectedPrimitive.asStringValue()}, actual=$actualPrimitivesValues"
```